pkg/app/pipedv1/plugin/kubernetes/deployment/pipeline.go

```diff
@@ -197,3 +198,38 @@ func buildPipelineStages(stages []*deployment.BuildPipelineSyncStagesRequest_Sta
 
 	return out
 }
+
+// buildPipelineStagesWithSDK builds the pipeline stages with the given SDK stages.
+// TODO: Rename this function to buildPipelineStages after removing the old one.
+func buildPipelineStagesWithSDK(stages []sdk.StageConfig, autoRollback bool) []sdk.PipelineStage {
+	out := make([]sdk.PipelineStage, 0, len(stages)+1)
+
+	for _, s := range stages {
+		out = append(out, sdk.PipelineStage{
+			Name:               s.Name,
+			Index:              s.Index,
+			Rollback:           false,
+			Metadata:           make(map[string]string, 0),
+			AvailableOperation: sdk.ManualOperationNone,
+		})
+	}
+
+	if autoRollback {
+		// we set the index of the rollback stage to the minimum index of all stages.
+		minIndex := slices.MinFunc(stages, func(a, b sdk.StageConfig) int {
+			return a.Index - b.Index
+		}).Index
+
+		s, _ := GetPredefinedStage(PredefinedStageRollback)
+		// we copy the predefined stage to avoid modifying the original one.
+		out = append(out, sdk.PipelineStage{
+			Name:               s.GetName(),
```

Comment:
The `s` is used only here, so can we replace this with `string(StageK8sRollback)`?

Reply:
OK, fixed on [d0fd69f](https://github.com/pipe-cd/pipecd/pull/5672/commits/d0fd69fefa7a92d0dabfdb0122ae6d8e6ca39bd5)
Also, I used `StageK8sRollback.String()` method.